examples/async.rs

```diff
@@ -116,10 +116,10 @@ fn main() {
             let session2 = client
                 .authenticate_device("testdeviceid2", None, true, HashMap::new())
                 .await;
-            let mut session = session.unwrap();
-            let mut session2 = session2.unwrap();
-            web_socket.connect(&mut session, true, -1).await;
-            web_socket2.connect(&mut session2, true, -1).await;
+            let session = session.unwrap();
+            let session2 = session2.unwrap();
+            web_socket.connect(&session, true, -1).await;
+            web_socket2.connect(&session2, true, -1).await;
         }
     };
 
```

Comment:
What's the state of the commented code below as well as the `Box.pin()` calls?

Reply:
I will remove the commented code as it is not necessary and simplified the code so that no `Box.pin` is necessary